src/query/server/server.go

```diff
@@ -409,19 +387,24 @@ func newDownsamplerAutoMappingRules(
 	return autoMappingRules, nil
 }
 
-func initClusters(cfg config.Configuration, dbClientCh <-chan client.Client, logger *zap.Logger) (local.Clusters, error) {
+func initClusters(
```

Comment:
isn't it better to initialize worker pools outside and pass them to this func ?

Reply:
This is for iterator pools, not the worker pools. If we're running embedded, we share the pools from m3db, so we need to pull them out after creating our session object here; if we're standalone, we spin up our own pools instead.